validator/db/kv/attester_protection.go

```diff
@@ -258,13 +294,28 @@ func (s *Store) batchAttestationWrites(ctx context.Context) {
 // and resets the list of batched attestations for future writes.
 // This function notifies all subscribers for flushed attestations
 // of the result of the save operation.
-func (s *Store) flushAttestationRecords(ctx context.Context) {
+func (s *Store) flushAttestationRecords(ctx context.Context, records []*AttestationRecord) {
+	if s.batchedAttestationsFlushInProgress {
```

Comment:
Atomic bool could be better here, we have it in the shared/ package

Reply:
abool

Reply:
Definitely, thanks!